UnityProject/Assets/Scripts/Weapons/Melee/MeleeStun.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Adding this to a weapon stuns the target on hit
+/// If the weapon has the StunBaton behaviour it only stuns when the baton is active
+/// </summary>
+public class MeleeStun : Interactable<HandApply>
+{
+	/// <summary>
+	/// How long to stun for (in seconds)
+	/// </summary>
+	public float stunTime;
```

Comment:
If this is public to set it in the editor I'd suggest doing this instead to make it more obvious:
```suggestion
	[SerializeField]
	private float stunTime;
```